data/topic-filter.go

```diff
@@ -0,0 +1,222 @@
+		ShowInWebUI:     true,
+	}
+
+	// Veterans - Search information specific for Veterans
+	Veterans = TopicContentType{
+		LocaliseKeyName: "Veteran",
+		Group:           "veterans",
+		Types:           []string{"veterans"},
+		ShowInWebUI:     true,
+	}
+
+	// topicFilterOptions contains all the possible filter available on the search page
+	topicFilterOptions = map[string]TopicContentType{
+		DemographyAndMigration.Group:                 DemographyAndMigration,
+		Education.Group:                              Education,
+		EthnicGroupNationalIdentityAndReligion.Group: EthnicGroupNationalIdentityAndReligion,
+		HealthDisabilityAndUnpaidCare.Group:          HealthDisabilityAndUnpaidCare,
+		Housing.Group:                                Housing,
+		LabourMarketAndTravelToWork.Group:            LabourMarketAndTravelToWork,
+		SexualOrientationAndGenderIdentity.Group:     SexualOrientationAndGenderIdentity,
+		Veterans.Group:                               Veterans,
+	}
+)
+
+// reviewFilter retrieves filters from query, checks if they are one of the filter options, and updates validatedQueryParams
+func reviewTopicFilters(ctx context.Context, urlQuery url.Values, validatedQueryParams *SearchURLParams) error {
+	topicFiltersQuery := urlQuery["topic"]
+
+	for _, topicFilterQuery := range topicFiltersQuery {
+
+		topicFilterQuery = strings.ToLower(topicFilterQuery)
+
+		if topicFilterQuery == "" {
+			continue
+		}
+
+		topicFilter, found := topicFilterOptions[topicFilterQuery]
+
+		if !found {
+			err := errs.ErrFilterNotFound
+			logData := log.Data{"topic filter not found": topicFilter}
+			log.Error(ctx, "failed to find topic filter", err, logData)
+
+			return err
+		}
+
+		validatedQueryParams.TopicFilter.Query = append(validatedQueryParams.TopicFilter.Query, topicFilter.Group)
+		validatedQueryParams.TopicFilter.LocaliseKeyName = append(validatedQueryParams.TopicFilter.LocaliseKeyName, topicFilter.LocaliseKeyName)
+	}
+
+	return nil
+}
+
+// GetTopicCategories returns all the categories and its content types where all the count is set to zero
+func GetTopicCategories() []TopicCategory {
+	var topicCategories []TopicCategory
+	topicCategories = append(topicCategories, TopicCategories...)
+
+	// To get a different reference of TopicContentType - deep copy
+	for i, topicCategory := range topicCategories {
+		topicCategories[i].ContentTypes = []TopicContentType{}
+		topicCategories[i].ContentTypes = append(topicCategories[i].ContentTypes, TopicCategories[i].ContentTypes...)
+
+		// To get a different reference of SubTypes - deep copy
+		for j := range topicCategory.ContentTypes {
+			topicCategories[i].ContentTypes[j].Types = []string{}
+			topicCategories[i].ContentTypes[j].Types = append(topicCategories[i].ContentTypes[j].Types, TopicCategories[i].ContentTypes[j].Types...)
+		}
+	}
+
+	return topicCategories
+}
+
+// updateQueryWithAPITopicFilters retrieves and adds all available sub filters which is related to the search filter given by the user
```

Comment:
```suggestion
// updateQueryWithAPITopicFilters adds all available subtopic filters to the API query
```

Reply:
Suggest changing this comment to the one I suggested above - clean and concise; no need to mention that a user inputs these filters

```suggestion
// updateQueryWithAPITopicFilters adds all available subtopic filters to the API query
```